app/utils/service_client/middleware/result_mapper.rb

```diff
@@ -0,0 +1,32 @@
+module ServiceClient
+  module Middleware
+    class ResultMapper < MiddlewareBase
+
+      def leave(ctx)
+        res = ctx.fetch(:res)
+
+        res_object =
+          if res.fetch(:success)
```

Comment:
res.fetch makes no sense because in both branches of if-else we rely on res being defined and call res[:status] etc. directly. So just `res[:success]` here.